docs/integration-guide.md

```diff
@@ -107,6 +107,239 @@ Here's how we recommend doing it:
+def api_helloworld_webhook(request, user_profile, client,
+        payload=REQ(argument_type='body'), stream=REQ(default='test'),
+        topic=REQ(default='Hello World')):
+
+  # construct the body of the message
+  body = ('Hello! I am happy to be here! :smile: ')
+
+  # add a wikipedia link if there is one in the payload
+  if ('type' in payload and payload['type'] == 'wikipedia'):
+      body += '\nThe Wikipedia featured article for today is **[%s](%s)**' % (payload['featured_title'], payload['featured_url'])
+
+  # send the message
+  check_send_message(user_profile, client, 'stream', [stream], topic, body)
+
+  # return json result
+  return json_success()
+
+```
+
+The above code imports the required functions, defines the main webhook
+function, decorating it `api_hey_only_webhook_view` and
+`has_request_variables`.
+
+You must pass the name of your webhook to the `api_key_only_webhook_view`
+decorator. Here we have used `HelloWorld`.
+
+You may name your webhook function whatever you like, though it's a good idea
+to be consistent with other webhook integrations. We recommend following the
+format `api_webhookname_webhook`.
+
+At minimum, the webhook function must accept `request`, `user_profile`, and
+`client`. You may also want to define additional parameters using the `REQ`
+object.
+
+In the example above, we have defined `payload` which is populated from the
+body of the http request, `stream` with a default of `test` (available by
+default in Zulip dev environment), and `topic` with a default of `Hello World`.
+
+In the body of the function we define the body of the message as `Hello! I am
+happy to be here! :smile: `. The `:smile:` indicates an emoji. Then we append a
+link to the Wikipedia article of the day as provided by the json payload.
+
+Then we send a public (stream) message with `check_send_message`.  Finally, we
+return `json_success()`.
+
+#### Step 2: Create an api endpoint for the webhook
+
+In order for a webhook to be externally available, it must be mapped to a url.
+This is done in `zproject/urls.py`. Look for the lines:
+
+```
+# Incoming webhook URLs
+urls += [
+    # Sorted integration-specific webhook callbacks.
+```
+
+And you'll find the entry for Hello World:
+
+```
+url(r'^api/v1/external/helloworld$',  'zerver.views.webhooks.helloworld.api_helloworld_webhook'),
+```
+
+This tells the Zulip api to call the `api_helloworld_webhook` function in
+`zerver/views/webhooks/helloworld.py` when it receives a request at
+`/api/v1/external/helloworld`.
+
+At this point, if you're following along and/or writing your own Hello World
+webhook, you have written enough code to test your integration.  You can do so
+by using curl on the command line:
+
+```
+curl -X POST -H "Content-Type: application/json" -d '{ "type":"wikipedia", "day":"Wed, 6/1", "featured_title":"Marilyn Monroe", "featured_url":"https://en.wikipedia.org/wiki/Marilyn_Monroe" }' http://localhost:9991/api/v1/external/helloworld\?api_key\=<api_key>
```

Comment:
It might be cleaner to have them create a fixture as step 0 (recommending requestb.in as a default approach), and then using `./manage.py send_webhook_fixture_message` to do the test, since that's probably lot more robust than giving them a curl command line with a bunch of escape characters in it :).